unattended.sh

```diff
@@ -1,178 +1,615 @@
+ fi
+ 
+ if [ "$(echo "${os_release}" | wc -l)" -ne 1 ]; then
+    error_and_exit "Unable to determine your running OS as there are multiple packages providing redhat-release : $os_release"
+ fi
 
-# systemd check.
-command -v systemctl > /dev/null 2>&1
-if [ "x$?" '=' x0 ] ; then
-  systemctl show > /dev/null 2>&1
-  if [ "x$?" '=' x0 ] ; then
-    has_systemd=1
-  else
-    has_systemd=0
-  fi
-else
-  has_systemd=0
-fi
+ os_version=$(rpm -q "${os_release}" --qf "%{version}")
 
-print_step_end
+ log "INFO" "Your running OS is $os_version"
 
+}
+#========= end of function get_os_release()
+
+
+#========= begin of function set_required_prerequisite()
+# check if the target OS is compatible with Red Hat and the version is 7 or 8
+# then set the required environment variables accordingly
 #
-# SELINUX
-#
+function set_required_prerequisite(){
 
-print_step_begin "SELinux deactivation"
-if [ -e /etc/selinux/config ] ; then
-  sed -i -e 's/^SELINUX=.*$/SELINUX=disabled/' /etc/selinux/config
-fi
-command -v selinuxenabled > /dev/null 2>&1
-if [ "x$?" '=' x0 ] ; then
-  selinuxenabled
-  if [ "x$?" '=' x0 ] ; then
-    setenforce 0
-    if [ "x$?" '!=' x0 ] ; then
-      error_and_exit "Could not disable SELinux. You might need to run this script as root."
-    fi
-    print_step_end
-  else
-    print_step_end "OK, already disabled"
-  fi
-else
-  print_step_end "OK, not detected"
-fi
+	log "INFO" "Check if the system OS is supported and set the environment variables"
+
+  get_os_release
+
+	case "$os_version" in
+    7*)
+      log "INFO" "Setting specific part for v7 ($os_version)"
+     
+      case "$os_release" in
+        centos-release* | centos-linux-release*)
+          BASE_PACKAGES=(centos-release-scl)
+          ;;
+        
+        oraclelinux-release*|enterprise-release*)
+          BASE_PACKAGES=(oraclelinux-release-el7)
+          ;;
+      esac
+
+      RELEASE_RPM_URL="http://yum.centreon.com/standard/$CENTREON_MAJOR_VERSION/el7/stable/noarch/RPMS/centreon-release-$CENTREON_RELEASE_VERSION.el7.centos.noarch.rpm"
+		  PHP_BIN="/opt/rh/rh-php72/root/bin/php"
+		  PHP_ETC="/etc/opt/rh/rh-php72/php.d"
+		  OS_SPEC_SERVICES="rh-php72-php-fpm httpd24-httpd"
+		  PKG_MGR="yum"
+		  CENTREON_REPO="centreon-$repo\*"
+
+      log "INFO" "Installing required base packages"
+      if ! $PKG_MGR -y -q install ${BASE_PACKAGES[@]}
+      then
+        error_and_exit "Failed to install required base packages  ${BASE_PACKAGES[@]}"
+      fi
+		  ;;
 
+    8*) 
+      log "INFO" "Setting specific part for v8 ($os_version)"
+      
+      case "$os_release" in
+        centos-release* | centos-linux-release*)
+          BASE_PACKAGES=(dnf-plugins-core epel-release)
+          dnf config-manager --set-enabled powertools
```

Comment:
there is an issue here with CentOS version before 8.3, the repository id `PowerTools`

Reply:
> there is an issue here with CentOS version before 8.3, the repository is `PowerTools` with upper cases

wouldn't it be interesting to apply (or recommend with a note) a `dnf update` before running the script routines ?